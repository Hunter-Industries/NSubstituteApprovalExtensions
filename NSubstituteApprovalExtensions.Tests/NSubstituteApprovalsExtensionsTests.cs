using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NSubstitute;
using NSubstituteApprovalsExtetensions;

namespace NSubstituteApprovalExtetensionsTests
{

    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class NSubstituteApprovalsExtensionsTests
    {

        [TestMethod]
        public void TestDefinedReturnValue()
        {
            var mockedDependency = Substitute.For<IDependencyInterface>();
            mockedDependency.Call1().Returns(10);

            TheLegacyCodeToTest(mockedDependency);

            mockedDependency.VerifyNSubstituteMockBehavior();
        }


        [TestMethod]
        public void TestNoPredefinedReturnValues()
        {
            var mockedDependency = Substitute.For<IDependencyInterface>();

            TheLegacyCodeToTest(mockedDependency);

            mockedDependency.VerifyNSubstituteMockBehavior();
        }

        [TestMethod]
        public void TestExampleOfWhatIUsedToDo()
        {
            var mockedDependency = Substitute.For<IDependencyInterface>();
            mockedDependency.Call1().Returns(10);

            TheLegacyCodeToTest(mockedDependency);

            Received.InOrder(() => {
                mockedDependency.Call1();
                mockedDependency.Call2("test");
                mockedDependency.Call1();
                mockedDependency.Call3("some more arguments", 4);
            });

            mockedDependency.DidNotReceive().NeverCalled();
        }

        private static void TheLegacyCodeToTest(IDependencyInterface legacyDependency)
        {
            var result = legacyDependency.Call1();
            legacyDependency.Call2("test");
            legacyDependency.Call1();
            legacyDependency.Call2("test" + result);

            if (result > 4)
                legacyDependency.Call3("some more arguments", 4);
            else
                legacyDependency.Call4(new ClassWithMultiLineToString());
        }
    }
}