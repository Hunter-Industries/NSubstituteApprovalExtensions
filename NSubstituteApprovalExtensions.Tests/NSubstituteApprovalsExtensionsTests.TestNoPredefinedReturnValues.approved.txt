Recorded behavior for: IDependencyInterface
Method Name: Call1
Arguments: 


Method Name: Call2
Arguments: 
	Type: String
	Value:
		test


Method Name: Call1
Arguments: 


Method Name: Call2
Arguments: 
	Type: String
	Value:
		test0


Method Name: Call4
Arguments: 
	Type: ClassWithMultiLineToString
	Value:
		ClassWith
		MultiLine
		To
		String


